src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -762,7 +773,24 @@ private async Task RegisterAsStreamProducer(StreamId streamId, StreamSequenceTok
                 if (pubSub == null) throw new NullReferenceException("Found pubSub reference not set up correctly in RetreaveNewStream");
 
                 IStreamProducerExtension meAsStreamProducer = this.AsReference<IStreamProducerExtension>();
-                ISet<PubSubSubscriptionState> streamData = await pubSub.RegisterProducer(streamId, streamProviderName, meAsStreamProducer);
+                ISet<PubSubSubscriptionState> streamData = null;
+                var callPubsubSucceed = false;
+                while (!callPubsubSucceed)
+                {
+                    var retryInterval = TimeSpan.FromSeconds(1);
```

Comment:
Should this be configurable?

Reply:
I'm OK with both. we can add a setting to PullingAgentOptions.